nvtabular/workflow/workflow.py

```diff
@@ -311,6 +340,9 @@ def _transform_ddf(ddf, workflow_nodes, meta=None):
     if isinstance(workflow_nodes, WorkflowNode):
         workflow_nodes = [workflow_nodes]
 
+    if not workflow_nodes:
+        return ddf
```

Comment:
@rjzamora Also want to call out this change, because I'm not sure what the performance implications are here. Since we've refactored out the concept of an input node (that only has a list of columns to select), there are now cases where `_transform_ddf` gets called with an empty list of parent nodes. This check avoids the rest of the function struggling to handle that case and throwing errors.

However, it also means we bypass this chunk of code, which never executes anymore:
```
# Check if we are only selecting columns (no transforms).
# If so, we should perform column selection at the ddf level.
# Otherwise, Dask will not push the column selection into the
# IO function.
if all((c.op is None and not c.parents) for c in workflow_nodes):
    return ddf[_get_unique(columns)]
```
We're not sure if we need to explicitly do something to push the column selection down into the IO function now that we don't have input nodes, or if we can delete that chunk of code and call it good. Thoughts?

Reply:
Good question.  I'm pretty sure that we will need to apply this same column projection in a new way now that the column selection is represented differently.  For example, we may want to add an optional `input_columns=` argument to `_transform_ddf`, and always start with the `ddf[input_columns]` projection (when input_columns is known) in that function.  In `fit`, above, we would pass in the union of `addl_input_cols` and `workflow_node.selector`.

We definitely want to apply column projections on the origin ddf whenever possible, otherwise Dask will be unable to push the projection into `read_parquet`.

Note that I may be misunderstanding the selector/node machinery here, so feel free to correct me.

Reply:
That sounds right to me. We'll give it a try.

Reply:
@rjzamora I think this is along the lines of what you were thinking? We end up pushing the additional column fetch down into `_transform_partition`, where it can happen in the same way as the other column concats.

Reply:
As a result, the check to see if we're only selecting columns without transforms can move up to the top of this method so that we don't skip over it when there are no parents to pull output columns from.